src/ProjectTemplates/Shared/ProcessLock.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Templates.Test.Helpers
+{
+    public class ProcessLock
+    {
+        public static readonly ProcessLock DotNetNewLock = new ProcessLock("dotnet-new");
+        public static readonly ProcessLock NodeLock = new ProcessLock("node");
+
+        public ProcessLock(string name)
+        {
+            Name = name;
+            Semaphore = new SemaphoreSlim(1);
```

Comment:
Why don't you use a spinlock instead? It already has all the asyncness you're looking for.

Reply:
The current code was already using a SemaphoreSlim (https://github.com/dotnet/aspnetcore/pull/21558/files#diff-a751368b99293f26ab0b48c583add848L36), I didn't want to change the semantics of it. Just wanted to make sure we timed out

Reply:
Fair enough.